x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/index.tsx

```diff
@@ -178,7 +180,14 @@ const AttackDiscoveryPageComponent: React.FC = () => {
 
   const pageTitle = useMemo(() => <PageTitle />, []);
 
-  const { sourcererDataView } = useSourcererDataView();
+  let { sourcererDataView } = useSourcererDataView();
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  const { dataViewSpec } = useDataViewSpec();
+
+  if (newDataViewPickerEnabled) {
+    sourcererDataView = dataViewSpec;
```

Comment:
Instead of rewriting the value a cleaner approach would create two constants `old...DataView` and `new...DataView` and assign to the constant `sourcererDataView` in a tertiary operator like

```ts
const sourcererDataView = newDataViewPickerEnabled ? `new...DataView` : `old...DataView`;
```

It significantly increases the locality.

**This comment is applicable to the similar code chunks in multiple files.** I'd refrain from repeating it over.

Reply:
I've considered that but instead of one if statement I would end up with multiple ternaries... it's also not like we will have it in there forever, probably for a month or less (until the previous code is removed).

Reply:
anyway, did it just because I could :D